src/mad_spark_alt/core/intent_detector.py

```
@@ -0,0 +1,353 @@
+            # Don't expect all keywords to match, just a few good ones
+            if score > 0:
+                # Normalize based on actual matches, not total possible
+                normalized_score = min(score / 5.0, 1.0)  # 5 matches = 100% confidence
+            else:
+                normalized_score = 0.0
+
+            intent_scores[intent] = (normalized_score, matched_keywords)
+
+        # Sort intents by score
+        sorted_intents = sorted(
+            intent_scores.items(), key=lambda x: x[1][0], reverse=True
```

Ruff (0.12.2)

Trailing comma missing

Add trailing comma

(COM812)